src/rust/engine/process_executor/src/remote.rs

```diff
@@ -1,23 +1,57 @@
 use std::error::Error;
+use std::sync::Arc;
 
 use bazel_protos;
 use crypto::digest::Digest;
 use crypto::sha2::Sha256;
+use grpcio::{ChannelBuilder, Environment};
 use protobuf;
+use protobuf::Message;
 
 use super::{ExecuteProcessRequest, ExecuteProcessResult};
 
 ///
 /// Runs a command via a gRPC service implementing the Bazel Remote Execution API
 /// (https://docs.google.com/document/d/1AaGk7fOPByEvpAbqeXIyE8HX_A3_axxNnvroblTZ_6s/edit).
 ///
-pub fn run_command_remote(req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+pub fn run_command_remote(addr: &str, req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+  let execute_request = make_execute_request(&req)?;
+
+  let env = Arc::new(Environment::new(1));
+  let channel = || {
+    ChannelBuilder::new(Arc::clone(&env)).connect(addr)
+  };
+  let execution_client = bazel_protos::remote_execution_grpc::ExecutionClient::new(channel());
+
+  let result = execution_client.execute(execute_request).unwrap();
+
+  // TODO: Check whether there's a response on this result.
+
+  let operation_client = bazel_protos::operations_grpc::OperationsClient::new(channel());
+  let mut operation_request = bazel_protos::operations::GetOperationRequest::new();
```

Comment:
Take it or leave it: where possible, I like to use a temporary scope to drop mutability once it is no longer necessary
```
let operation_request = {
  let mut or  = bazel_protos::operations::GetOperationRequest::new();
  or.set_name(result.get_name().to_string());
  or
};
```

Reply:
I've done that each place that more than one field is set on the proto, but it seemed like overkill to double the length of the definition, particularly as the very next line moves the variable. Maybe consistency is better, though...